Libraries/LinkingIOS/RCTLinkingManager.m

```diff
@@ -47,7 +47,20 @@ + (BOOL)application:(UIApplication *)application
   sourceApplication:(NSString *)sourceApplication
          annotation:(id)annotation
 {
-  NSDictionary<NSString *, id> *payload = @{@"url": URL.absoluteString};
+  NSURLComponents *urlComponents = [NSURLComponents componentsWithURL:URL
+                                              resolvingAgainstBaseURL:NO];
+  NSMutableDictionary<NSString *, NSString *> *parameters = [NSMutableDictionary new];
+  for (NSURLQueryItem *queryItem in urlComponents.queryItems) {
+    parameters[queryItem.name] = queryItem.value;
+  }
+
+  NSDictionary<NSString *, id> *payload = @{
+    @"host": urlComponents.host,
```

Comment:
These properties are all optional (may be nil). You should probably suffix them all with ?: @"", or wrap them in RCTNullIfNil(...) if null would be preferable on the JS side.